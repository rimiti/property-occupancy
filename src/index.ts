export * from './lib/occupancy';
export * from './lib/occupancy/types';
