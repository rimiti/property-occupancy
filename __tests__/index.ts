import MockDate from 'mockdate';
import { getOccupancyRateByDay, getOccupancyRateByMonth, TOccupancyRateParams } from '../src';

describe('getOccupancyRateByDay', () => {
  beforeAll(() => {
    MockDate.set('2024-01-01T00:00:00Z');
  });

  afterAll(() => {
    MockDate.reset();
  });

  const params: TOccupancyRateParams = {
    reservations: [
      {
        checkInAt: new Date('2023-05-01T00:00:00Z'),
        checkOutAt: new Date('2023-05-03T00:00:00Z'),
      },
      {
        checkInAt: new Date('2023-05-04T00:00:00Z'),
        checkOutAt: new Date('2023-05-06T00:00:00Z'),
      },
    ],
    startDate: new Date('2023-05-01T00:00:00Z'),
    endDate: new Date('2023-05-06T00:00:00Z'),
    numberOfProperties: 10,
  };

  it('should return the correct occupancy rate by day', () => {
    const result = getOccupancyRateByDay(params);

    expect(result).toMatchSnapshot();
  });
});

describe('getOccupancyRateByMonth', () => {
  beforeAll(() => {
    MockDate.set('2025-01-01T00:00:00Z');
  });

  afterAll(() => {
    MockDate.reset();
  });

  test('returns correct occupancy rates by month', () => {
    const params = {
      reservations: [
        { checkInAt: new Date('2022-01-01'), checkOutAt: new Date('2022-01-03') },
        { checkInAt: new Date('2022-01-02'), checkOutAt: new Date('2022-01-04') },
        { checkInAt: new Date('2022-02-01'), checkOutAt: new Date('2022-02-03') },
        { checkInAt: new Date('2022-02-02'), checkOutAt: new Date('2022-02-04') },
      ],
      startDate: new Date('2022-01-01'),
      endDate: new Date('2022-02-28'),
      numberOfProperties: 2,
    };

    const occupancyRates = getOccupancyRateByMonth(params);

    expect(occupancyRates).toMatchSnapshot();
  });
});
