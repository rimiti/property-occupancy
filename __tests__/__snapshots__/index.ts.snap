// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOccupancyRateByDay should return the correct occupancy rate by day 1`] = `
[
  {
    "date": "2023-05-01",
    "occupancyRate": 10,
  },
  {
    "date": "2023-05-02",
    "occupancyRate": 10,
  },
  {
    "date": "2023-05-03",
    "occupancyRate": 0,
  },
  {
    "date": "2023-05-04",
    "occupancyRate": 10,
  },
  {
    "date": "2023-05-05",
    "occupancyRate": 10,
  },
  {
    "date": "2023-05-06",
    "occupancyRate": 0,
  },
]
`;

exports[`getOccupancyRateByMonth returns correct occupancy rates by month 1`] = `
[
  {
    "date": "2022-01",
    "occupancyRate": 6.45,
  },
  {
    "date": "2022-02",
    "occupancyRate": 7.14,
  },
]
`;
